### ISSUE TITLE
Features Request

### ISSUE THREAD

**Issue by tarekbazine** [22fevrier2019/22fevrier2019#1]
First, I would like to thank you for this initiative -good job; Second, I have some propositions:
- [ ] We should think (discuss) about a mechanism to identify the voter so that we could avoid bots and duplicate votes.
- [ ] categories or tags. 
- [ ] a mobile app (just throw me the ui design and i could help with it)

**Comment by MounirB** [comment]
Hello,
I agree with the idea of tags and categories, and the mobile app. However, I think that the privacy of the voter/proposer is more important than the risk of duplicate votes

**Comment by TaySai** [comment]
I agree with your argument @MounirB  that privacy is more important than the risk of duplication.
In my case, I would vote, but when I saw that I had to connect with facebook or google, I simply refused.
Maybe a google reCaptcha can solve your problem?

**Comment by Ozsiix** [comment]
I started working on the same project 4 days ago, but looks like you guys beat me to it ,, so here is what i have done so far, 

-  I categorized the messages with " Niveaux scolaire " &  " profession "
-  Made the UI responsive so it can be opened on phones & tablets  ( i was gonna develop an android & iOS app later on )
-  pagination feature on the home page so it won't load +1000 messages in one page 
-  Made the UI design look more patriotic. 
-  i was currently working on comments section for each message so people can discuss one message instead of getting more repetitive messages.  

Plus i totally agree with @terryBaz on avoiding bots thing from a security and credibility POV .

**Comment by mavrickdeveloper** [comment]
i suggest implementing the Blockchain technology in voting mechanism which will make the voting virtually impossible to fake
thoughts ?

**Comment by tarekbazine** [comment]
_thank you @lightingcoders for the suggestion (thanks all of you for this discussion)_

Blockchain sounds great ! all though **(for me) I** do not have a clue on **how to put this technology in real world (in the context of this project)** and also for the technicals needs such as **infras** ..etc, any suggestion or **concrete explanation of the implementation**, will be a huge help.

**Comment by Hamzandj** [comment]
Salam alaikoum, Hello everybody. I want to thank you for the good job you're doing,and wanted to give some ideas to help : 
- [ ] Adding a home page to explore subjects by Theme and sub categories (based on the tags you have already added).
- [ ] Propose some forms to answer some questions, kind of a formalisation of the most commented and interesting ideas.
- [ ] Add research filters on each category to know if a the contribution, the person wants to post, have already been posted.

you can refer to a similar [website](https://granddebat.fr) made by the French government to deal with popular claims so you can maybe take some ideas .

**Comment by bmole31** [comment]
Hi everybody ! I d like to thank you so much for this initiative that confirms that Algeria has great talents who need to be free. 
I am right now studying bioinformatics in Montpellier, so my informatical culture is a bit limited (more biologist than informatican) but I can call for some suggestions and help from my fellows. 
I d also like to give some propositions about the website without being redundant:
A welcoming page to explain the initiative and to give links To the différents subjects. There a nice example of this kind of website here: https://granddebat.fr
I don't know if it's really possible to use a translator to translate all the propositions in French to Arabic and vice versa to get a liguistical homogeneous website.
That's all for me guy's, looking forward to reading your feedbacks
Mourad

**Comment by bidossessi** [comment]
Hi all. Some great contributions here.

Totally agree for the home page. It is essential. I believe users should be encouraged to make **one** and only one proposal per item. It would ease referral. I know it's not possible to enforce it, but a good home page could help in directing users as to how it would be beneficial.

@Ozsiix I understand your idea of avoiding redundancy, but I think existing social networks can be leveraged to deal with the "discussion" part, as long as the platform has proper sharing and permalinks. It would be a serious overhead to add it here.

There should probably be a feature to "merge" similar proposals, and consolidate their support/opposition. I'm still thinking of how it could look visually.
